common/changes/@typespec/versioning/fix-versioning-across-namespaces_2023-03-31-18-34.json

```diff
@@ -0,0 +1,15 @@
+{
+  "changes": [
+    {
+      "packageName": "@typespec/versioning",
+      "comment": "Fix: Issue with using version as a template parameter across different namespace. Include a significant change in the versioning library internals.",
+      "type": "none"
+    },
+     {
+      "packageName": "@typespec/versioning",
+      "comment": "**API BREAKING** `getAvailabilityMap` is now returning a `Map<TimelineMoment, Availability>` instead of `Map<Version, Availability>`.",
```

Comment:
I'm seeing `getAvailablilityMap` returning the same thing as before. There's a new `getAvailabilityMapInTimeline` that has the new signature. The breaking change seems to be `indexVersions` -> `indexTimeline`

Reply:
ah yeah I forgot I didn't change it because it was making a big change to the validation logic and will keep that as a seperate change as it has many issues already but those are not blocking

So there isn't any breaking changes then.

Reply:
hhm, seems like this uncovered an issue that versioning of dep of dep never worked.